include/platform.h

```diff
@@ -27,6 +27,15 @@
 # define S_ISDIR(mode) (((mode) & S_IFMT) == S_IFDIR)
 #endif
 
+// gcc has __PRETTY_FUNCTION__, MSVC has __FUNCSIG__, __func__ is standard
```

Comment:
Even if we don't merge the C++ PR before the next release, this seems worth doing.